### PR TITLE
Named mutexes: fix handle leak, add stress tests

### LINKED ISSUE #5364: When creating or opening named mutex fails, handle leaks

### PR DESCRIPTION
**Pull request by kouvel**
Fix handle leak when creating or opening named mutex fails
Fixes #5364
- After the pal object is registered, it gets a second reference from the handle. Upon an error after that, the handle needs to be revoked in addition to releasing the initial reference.

Add named mutex stress tests
- The existing PAL tests are run in parallel, each in a separate thread, for a specified duration

### CODE REVIEW

**Comment by kouvel** [comment]
@sergiy-k @janvorli

File: src/pal/src/synchobj/mutex.cpp [hunk 1]
```diff
@@ -347,25 +355,20 @@ CorUnix::InternalCreateMutex(
...
-    if (NULL != pobjMutex)
+    _ASSERTE(pobjRegisteredMutex == nullptr);
+    if (hMutex != nullptr)
     {
-        pobjMutex->ReleaseReference(pthr);
+        g_pObjectManager->RevokeHandle(pthr, hMutex);
     }
-
-    if (NULL != pobjRegisteredMutex)
+    else if (NULL != pobjMutex)
```
Comment (adityamandaleeka) [inline src/pal/src/synchobj/mutex.cpp]: nullptr?

Comment (adityamandaleeka) [inline src/pal/src/synchobj/mutex.cpp]: Also, can you change the order to `pobjMutex != nullptr` so it's consistent with the previous if?

File: src/pal/src/synchobj/mutex.cpp [hunk 2]
```diff
@@ -712,25 +723,20 @@ CorUnix::InternalOpenMutex(
...
```
Comment (adityamandaleeka) [inline src/pal/src/synchobj/mutex.cpp]: Same comments as above.

**Comment by adityamandaleeka** [comment]
BTW, while you're here, can you look for 'phEvent' in mutex.cpp and fix the two occurrences in the parameter descriptions?

**Comment by adityamandaleeka** [comment]
LGTM modulo minor comments.

**Comment by sergiy-k** [comment]
Let's keep the changes to a required minimum. No clean up please. We most likely will need to port this to the release branch.

**Comment by kouvel** [comment]
@dotnet-bot test this please

**Comment by kouvel** [comment]
@adityamandaleeka thanks for the review, I'll need to make a cleanup pass over this feature later, I'll make the changes you suggested as part of that.

**Comment by kouvel** [comment]
@sergiy-k I plan to port only the first commit, which has the minimal required changes to fix the handle leak

**Comment by adityamandaleeka** [comment]
:+1:

**Pull request merged**